app/src/main/java/com/codesoom/assignment/CrudHttpHandler.java

```diff
@@ -0,0 +1,143 @@
+
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+
+        String method = exchange.getRequestMethod();
+        URI uri = exchange.getRequestURI();
+
+        String path = uri.getPath();
+        String id = getId(path);
+        OutputStream outputStream = exchange.getResponseBody();
+
+
+        String response = "[]";
+        /**
+         * method에 null이 들어온다면 비교주체자가 null이 되버리기 때문에 equals를 실행할 수 없어 NPE가 발생할 가능성이 생기는 것 같습니다.
+         * 반대로 "GET".equals(method) 로 변경하면 비교하는 주체가 null이 발생할 일이 없어지기 때문에 NPE 방지가 되는 원리 같습니다.
+         */
+        if("GET".equals(method) && ( ("/task".equals(path)) || ("/task/"+id).equals(path) ) ) {
+
+            if(!"noId".equals(id)){
+                Task task = tasks.get(Long.parseLong(id));
+                if(task==null)
+                    response = "아이디가 없습니다.";
+                else
+                    response = toTaskJsonOne(task);
+            }
+            else if(!tasks.isEmpty()) {
+                response = toTaskJson();
+            }
+
+            exchange.sendResponseHeaders(200,response.getBytes().length);
+        }
+        else if("POST".equals(method) && "/task".equals(path)) {
+
+            String content = getContent(exchange);
+            Task task = toTask(content);
+            task.setId(autoId++);
+            tasks.put(task.getId(), task);
+
+            response=toTaskJsonOne(task);
+            exchange.sendResponseHeaders(201,response.getBytes().length);
+        }
+        else if("PUT".equals(method) && ("/task/"+id).equals(path)) {
+
+            String content = getContent(exchange);
+            Task taskPut = toTask(content);
+
+            if(tasks.get(Long.parseLong(id)) == null)
+                response = "아이디가 없습니다.";
+            else{
+                Task task = tasks.get(Long.parseLong(id));
+                task.setTitle(taskPut.getTitle());
+                response=toTaskJsonOne(task);
+            }
+
+            exchange.sendResponseHeaders(200,response.getBytes().length);
+
+        }
+        else if("DELETE".equals(method) && ("/task/"+id).equals(path)) {
+
+            if(tasks.get(Long.parseLong(id)) == null)
+                response="아이디가 없습니다.";
+            else
+                tasks.remove(Long.parseLong(id));
+
+            response="DELETE..";
+            exchange.sendResponseHeaders(200,response.getBytes().length);
+        }
+        else{
+            /**
+             * 서버가 요청을 완료했다는 뜻을 담고 있습니다.
+             * https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.1
+             */
+            exchange.sendResponseHeaders(200, 0);
+        }
+
+
+        outputStream.write(response.getBytes());
+        outputStream.flush();
+        outputStream.close();
+    }
+
+    private String getId(String path) {
+
+        if(path.indexOf("/task/")==0) {
+            return path.replace("/task/","");
+        }
+        else {
+            return "noId";
+        }
+    }
```

Comment:
`}` 뒤에 바로 `else`를 붙여 주세요. 

```suggestion
    private String getId(String path) {

        if(path.indexOf("/task/") == 0) {
            return path.replace("/task/", "");
        } else {
            return "noId";
        }
    }
```

`"noId"`를 사용하는 방식은 기발하지만 최고의 방법은 아닙니다. 다른 방법도 생각해보세요. 좋은 훈련이 될 거에요.